overrides/default-settings.gschema.override

```diff
@@ -109,7 +109,6 @@ enable-adblock=false
 enable-smooth-scrolling=true
 
 [org.gnome.mutter]
-auto-maximize=false
```

Comment:
Mutter maximizes windows that are >80% of the display size, which breaks window state saving when a user intentionally does not maximize a window. Maximization should *always* be user-driven because it hides UI like the dock.

Reply:
@cassidyjames If I expand a window's size to >80% of the display size mutter does not maximize it. However when I open a window with that condition it's maximized.
Is that not intuitive?

> Maximization should always be user-driven because it hides UI like the dock.

Can you explain why this is an "always" aside from obscuring UI?
(it's less hidden and more obscured, I can still hover to trigger it's manifestation)

Reply:
Apps should save their state. Closing an app in a specific state and then reopening it should restore it to the state at which it was closed. By trying to be extra clever and intuit whether or not a window should be maximized, we're introducing spooky action that the user did not perform and may not have wanted, with no explanation. It's just not a pattern that we want in elementary OS.

Reply:
> Apps should save their state. Closing an app in a specific state and then reopening it should restore it to the state at which it was closed. By trying to be extra clever and intuit whether or not a window should be maximized, we're introducing spooky action that the user did not perform and may not have wanted, with no explanation. It's just not a pattern that we want in elementary OS.
> 
> Apps already can (and do!) save/restore maximization state. That's up to the apps and we should not paper over that at the DE level.

That's a sufficient explanation. TBH in elementary OS, this seems more like a behavior in mutter you could expose in an option. It's possible someone might appreciate that functionality.
Thoughts @cassidyjames? I do agree this shouldn't be a default.

Reply:
Eh, I think the behavior is a design decision for the platform to make, which we've made. It always seemed like a workaround for crappy app behaviors in Mutter, and it's so nuanced that it would take more effort to explain in the UI (and then perpetually support when thinking about the design of Pantheon moving forward) that it's not worth it imho.

Reply:
Right, it's extremely nuanced. I personally wouldn't want to explain that to anyone ever.